src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,194 @@
+        int oldCapacity = (oldTable == null) ? 0 : oldTable.length;
+        int oldThreshold = threshold;
+        int newCapacity;
+        int newThreshold = 0;
+
+        if (oldCapacity > 0) {
+            if (oldCapacity >= MAXIMUM_CAPACITY) {
+                threshold = Integer.MAX_VALUE;
+                return oldTable;
+            } else if ((newCapacity = oldCapacity << 1) < MAXIMUM_CAPACITY
+                    && oldCapacity >= DEFAULT_INITIAL_CAPACITY) {
+                newThreshold = oldThreshold << 1;
+            }
+        } else if (oldThreshold > 0) {
+            newCapacity = oldThreshold;
+        } else {
+            newCapacity = DEFAULT_INITIAL_CAPACITY;
+            newThreshold = (int) (DEFAULT_LOAD_FACTOR * DEFAULT_INITIAL_CAPACITY);
+        }
+
+        if (newThreshold == 0) {
+            float resizeThreshold = (float) newCapacity * 0.75f;
+            newThreshold = (newCapacity < MAXIMUM_CAPACITY
+                    && resizeThreshold < (float) MAXIMUM_CAPACITY
+                    ? (int) resizeThreshold : Integer.MAX_VALUE);
+        }
+        threshold = newThreshold;
+
+        @SuppressWarnings({"rawtypes", "unchecked"})
+        Node<K, V>[] newTable = (Node<K, V>[]) new Node[newCapacity];
+        table = newTable;
+        if (oldTable != null) {
+            for (int j = 0; j < oldCapacity; ++j) {
+                Node<K, V> element;
+                if ((element = oldTable[j]) != null) {
+                    oldTable[j] = null;
+                    do {
+                        Node<K, V> next = element.next;
+                        int index = element.hash & (newCapacity - 1);
+                        element.next = newTable[index];
+                        newTable[index] = element;
+                        element = next;
+                    } while (element != null);
+                }
+            }
+        }
+        return newTable;
+    }
 
     @Override
     public void put(K key, V value) {
+        Node<K, V>[] tableArray;
+        Node<K, V> currentNode;
+        int currentCapacity;
+        int index;
+
+        if ((tableArray = table) == null || (currentCapacity = tableArray.length) == 0) {
+            currentCapacity = (tableArray = resize()).length;
+        }
+
+        index = (currentCapacity - 1) & hash(key);
+        currentNode = tableArray[index];
+
+        if (currentNode == null) {
+            tableArray[index] = new Node(hash(key), key, value, null);
+            tableSize++;
+        } else {
+            Node<K, V> element;
+            K existingKey;
 
+            if (currentNode.hash == hash(key) && ((existingKey = currentNode.key) == key
+                    || (key != null && key.equals(existingKey)))) {
+                currentNode.value = value;
+            } else {
+                for (; ; ) {
+                    if ((element = currentNode.next) == null) {
+                        currentNode.next = new Node<>(hash(key), key, value, null);
```

Comment:
Remember to parameterize the Node class. Use `new Node<K, V>(...)` instead of `new Node<>(...)` to ensure type safety.